### ISSUE TITLE
Forcing curses upon companions at shrine

### ISSUE THREAD

**Issue by emmeken** [FloricSpacer/AbyssDiver#8]
On layer 3 there is the Skewed Shrine, which can  transfer curses to companions. On layer 4 there is the Steady Shrine, which can copy curses to companions. In this issue, I want to discuss the implementation of the layer 3 shrine, although whatever we decide will affect the layer 4 implementation as well.

Currently the Layer 3 Wonders passage asks the player to make the choice between a willing or unwilling transfer. I think it would make sense to instead travel to the shrine before having to make the decision about whether the transfer is going to be forced or not. In particular, I think it would be good to show the curses that companions will accept voluntarily before having the player consider forcing curses upon them.

Once at the shrine, the player would have the ability to transfer any number of curses with a companion's approval, provided they can afford the increasing dubloon costs. At any point, the player can decide to leave or to switch to forced curse transfers.

The current narration for forced curse transfers is a bit awkward. For example, the narration speaks of an incapacitated companion before the player has picked a victim. Also the player character gets negative reactions from all companions, but is not hindered when transferring another curse without permission.

I have some doubts that incapacitating companions is the way in which the CYOA intended curses to be forced. The text reads:

> With the range of the altar and how long it takes to operate, you should be able to get 4 Curse transfers to an unwilling participant before they can escape, if that's how you're playing.

This suggests to be that forcing one curse would cause a companion to run away and that there is enough time to force three more curses upon them before they've run out of reach of the shrine's power.

With that interpretation, the affection penalty for forcing a curse should be a lot higher than it currently is: receiving a single unwanted curse would be enough for the average companion to leave. Other companions witnessing this but not receiving curses would suffer an affection penalty but wouldn't leave immediately. Would they still be willing to accept curses they don't hate or is that option cut off as soon as a non-willing transfer is made?

Currently companions will block travelling to the shrine again if too many curses were forced. I like this idea, but maybe it should be decided using the affection score instead of using an explicit transfer counter.

What happens when companions leave the player character? The intro describes that they should not have problems ascending from the Abyss, so I guess they would go back to Outset Town (except for Cherry, perhaps). But if a companion is very upset with the player character, it wouldn't make sense to be able to re-hire them, so we should probably block that. Would this be done on an individual basis, or would one companion leaving damage the player character's reputation so much that no companion will want to work for them anymore?

**Comment by Ilyza91** [comment]
>Currently the Layer 3 Wonders passage asks the player to make the choice between a willing or unwilling transfer. I think it would make sense to instead travel to the shrine before having to make the decision about whether the transfer is going to be forced or not. In particular, I think it would be good to show the curses that companions will accept voluntarily before having the player consider forcing curses upon them.

I agree with this one

>The current narration for forced curse transfers is a bit awkward. For example, the narration speaks of an incapacitated companion before the player has picked a victim. Also the player character gets negative reactions from all companions, but is not hindered when transferring another curse without permission.

Yeah I see the point here, but this will be debatable. Will khemia or maybe even lily stop you when transfering curses? Yeah probably. For cloud that's already more debatable and I think cherry is really unlikely. That's my opinion and I imagine everybody opinion on this will be different

>With that interpretation, the affection penalty for forcing a curse should be a lot higher than it currently is: receiving a single unwanted curse would be enough for the average companion to leave. Other companions witnessing this but not receiving curses would suffer an affection penalty but wouldn't leave immediately. Would they still be willing to accept curses they don't hate or is that option cut off as soon as a non-willing transfer is made?

I always interpreted as, they run away to a safe distance. Ready to give you an earfull when you catch up with them again. But I see how you can interpret it that way as well. @FloricSpacer what do you think?

>Currently companions will block travelling to the shrine again if too many curses were forced. I like this idea, but maybe it should be decided using the affection score instead of using an explicit transfer counter.

I like this idea


> What happens when companions leave the player character? The intro describes that they should not have problems ascending from the Abyss, so I guess they would go back to Outset Town (except for Cherry, perhaps). But if a companion is very upset with the player character, it wouldn't make sense to be able to re-hire them, so we should probably block that. Would this be done on an individual basis, or would one companion leaving damage the player character's reputation so much that no companion will want to work for them anymore?

I would imagine they wouldn't want to be hired by you again. While I do see how you could get a negative reputation and some people would not want to work with you again, the opposite point can also be made that they don't care, and in that case I think it would make sense to leave the option in (maybe a run-away character could contribute to a negative score for unhired companions instead?)

**Comment by FloricSpacer** [comment]
I initially interpreted the forced use of shrines involving incapacitating the victim, then forcing the curses on them, however, I think the interpretation you two share makes more sense given the way it was written and I definitely like it better! So we should rewrite them with your interpretation (running out of range) in mind. 

> I think it would make sense to instead travel to the shrine before having to make the decision about whether the transfer is going to be forced or not. In particular, I think it would be good to show the curses that companions will accept voluntarily before having the player consider forcing curses upon them.

That also seems like a good idea to me, feel free to add it if you'd like to.

> With that interpretation, the affection penalty for forcing a curse should be a lot higher than it currently is: receiving a single unwanted curse would be enough for the average companion to leave. Other companions witnessing this but not receiving curses would suffer an affection penalty but wouldn't leave immediately. Would they still be willing to accept curses they don't hate or is that option cut off as soon as a non-willing transfer is made?`

Agreed, I think it should be enough that unless you formed a strong bond they'd immediately leave after 1 forced curse, then nearly anyone would leave after a few more. I also think that none of the companions would be willing to get close to the shrine again if you've used it in a forced way before, so maybe we can block that off after 1 use with a note saying no one will trust you enough to come to the shrine with you.

> I would imagine they wouldn't want to be hired by you again. While I do see how you could get a negative reputation and some people would not want to work with you again, the opposite point can also be made that they don't care, and in that case I think it would make sense to leave the option in (maybe a run-away character could contribute to a negative score for unhired companions instead?)

I think to keep it simple and understandable to players, maybe we could have it just block the hiring of the same companion again if they ever leave you due to affection dropping too low. We can say that they aren't close to the other available companions or the others are skeptical of their stories, but the one who left will never again adventure with you. I suspect having a separate "unhired companion reputation score" sort of system may seem opaque and confusing to players. What do you guys think of that?

> Yeah I see the point here, but this will be debatable. Will khemia or maybe even lily stop you when transfering curses? Yeah probably. For cloud that's already more debatable and I think cherry is really unlikely. That's my opinion and I imagine everybody opinion on this will be different

This does seem reasonable to me, different companions certainly would react differently to that. We could say that the shrine is quick to operate and has a short range, so there isn't much time to fight you, as you can quickly transfer a few curses before the companion gets out of the range quickly. That would preserve the ability to transfer 4 forced curses without needing to do anything based on the player's specific party composition.

**Comment by Ilyza91** [comment]
And after you force 1 to 4 curses on somebody, nobody want to accompany you to the shrine anymore? I think that's easy enough to implement and it also avoids the problem which character would stick their neck out to prevent you from doing it again.

**Comment by FloricSpacer** [comment]
> And after you force 1 to 4 curses on somebody, nobody want to accompany you to the shrine anymore? I think that's easy enough to implement and it also avoids the problem which character would stick their neck out to prevent you from doing it again.

Yep, that's what I was thinking. We can also warn players that if they force a curse on anyone, then no one will accompany them to the shrine in the future.

**Comment by emmeken** [comment]
On the [TFGS forum](https://tfgames.site/phpbb3/viewtopic.php?p=697576#p697576), DLDracoRex mentions that they would prefer if the CYOA restriction "A single given Curse can only be copied once - you cannot copy the same Curse option to multiple companions" were not implemented.

I don't know if there is a narrative or balance reason for the restriction. From an implementation perspective, not adding the restriction would be easier, since we wouldn't have to check whether a curse was copied before.

**Comment by FloricSpacer** [comment]
Yeah, I think it's probably fine to not implement that restriction.
